chia/server/rate_limits.py

```diff
@@ -43,7 +43,7 @@ def __init__(self, incoming: bool, reset_seconds: int = 60, percentage_of_limit:
 
     def process_msg_and_check(
         self, message: Message, our_capabilities: list[Capability], peer_capabilities: list[Capability]
-    ) -> bool:
+    ) -> tuple[bool, str]:
```

Comment:
Some smells, The string is always empty if the boolean is true.  If the boolean is false the string has content.  The string should perhaps be optional.  This would end up at `Union[tuple[Literal[True], None], tuple[Literal[False], str]]`.  But, perhaps just an `Option[str]` would cut it?  Basically, instead of returning 'success' like before we return 'maybe error message'.